daemon/algod/api/server/v2/handlers.go

```diff
@@ -785,6 +798,63 @@ func (v2 *Handlers) RawTransaction(ctx echo.Context) error {
 	return ctx.JSON(http.StatusOK, generated.PostTransactionsResponse{TxId: txid.String()})
 }
 
+// SimulateTransaction simulates broadcasting a raw transaction to the network, returning relevant simulation results.
+// (POST /v2/transactions/simulate)
+func (v2 *Handlers) SimulateTransaction(ctx echo.Context) error {
+	if !v2.Node.Config().EnableTransactionSimulator {
+		return ctx.String(http.StatusNotFound, "/transactions/simulate was not enabled in the configuration file by setting EnableTransactionSimulator to true")
+	}
+
+	stat, err := v2.Node.Status()
+	if err != nil {
+		return internalError(ctx, err, errFailedRetrievingNodeStatus, v2.Log)
+	}
+	if stat.Catchpoint != "" {
+		// node is currently catching up to the requested catchpoint.
+		return serviceUnavailable(ctx, fmt.Errorf("SimulateTransaction failed as the node was catchpoint catchuping"), errOperationNotAvailableDuringCatchup, v2.Log)
+	}
```

Comment:
Seems like this block is repeated 4-5 times in handlers.go.  Could you extract it to a function?

Reply:
@jdtzmn Optionally, I'd consider a higher order function to enforce the implied ordering.

If you want to pursue the idea, but it feels unclear, I'll happily elaborate live.

Reply:
Don't fight the Golang.

Reply:
@jannotti I'm genuinely _not_ trying to steer too far away from conventions.  I also don't want to distract the thread.  Perhaps live we can spend a few minutes here?

Probably my calibration of where it makes sense to apply the convention is off-base.

Unless messaged otherwise, let's disregard my (optional) remark RE: higher order function usage.

Reply:
Sure, let's talk at standup.  I'm mostly joking around, but I actually don't see the utility of a higher order function here. Those 7 lines are repeated verbatim (except for error message) about 5 times.  This seems like a job for drop dead simple:
```
func xxx(node NodeInterface, operation string)  (s node.StatusReport, err error) {
  // those 8 lines, with `operation` in the error message
}
```

Reply:
Resolving - Based on live discussion we agreed to keep as is.  HOF and/or a helper method run into more issues than they're worth right now.